Rename event args files to match the class name

hughbe opened the pull request:
Rename the files, make properties autoprops

All tested 100% coverage by https://github.com/dotnet/winforms/pull/274

zsd4yr commented:
@hughbe looks like there are a few conflicts to resolve. Once you've had a chance to look at those, I'll review the rest of these and hopefully we can get this in 😄

zsd4yr commented:
Could you provide some data backing up your 100% code coverage claim. What code coverage tool are you using?

sharwell commented:
I'm not especially happy about the scope of changes in commit 1ccd35c07dfec5b0a40fb683ac58e57ffe861724. I would have preferred to see each conceptual change as a separate commit. With the current form, it is impossible to state that the result is correct.

zsd4yr commented on src/System.Windows.Forms/src/System/Windows/Forms/BindingManagerDataErrorEventArgs.cs:
```diff
@@ -4,30 +4,13 @@
+            Exception = exception;
         }
+
+        public Exception Exception { get; }
```
are these code paths equivalent?

zsd4yr commented on src/System.Windows.Forms/src/System/Windows/Forms/CacheVirtualItemsEvent.cs:
```diff
@@ -1,30 +0,0 @@
-using System;
-namespace System.Windows.Forms {
-    /// <include file='doc\CacheVirtualItemsEventArgs.uex' path='docs/doc[@for="CacheVirtualItemsEventArgs"]/*' />
-    public class CacheVirtualItemsEventArgs : EventArgs {
```
Even though this may not be used in our codebase, this is a deletion of public API (see https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.cachevirtualitemseventargs?view=netframework-4.7.2) so deletion here may require api review. I will bring up to the team in order to get a definitive answer.

There are a couple of concerns here; on the top are
* is this used in other dotnet projects (especially WPF) 
  * https://referencesource.microsoft.com/#q=CacheVirtualItemsEventArgs is returning only results for the class declaration itself, so we're safe there
  * https://github.com/search?q=org%3Adotnet+CacheVirtualItemsEventArgs&unscoped_q=CacheVirtualItemsEventArgs is returning primarily docs and our source, which is good; however, there are **a couple of other results that are concerning**:
    1. use in dotnet/samples https://github.com/dotnet/samples/search?q=CacheVirtualItemsEventArgs&unscoped_q=CacheVirtualItemsEventArgs
    2. use in dotnet/project-system https://github.com/dotnet/project-system/search?q=CacheVirtualItemsEventArgs&unscoped_q=CacheVirtualItemsEventArgs 
  * there is another internal check I can do for this called _index_; which matches the results for reference source in this case
* is this used in Visual Studio (you cannot perform this check, but searching through VS source reveals **nothing of concern**)
* is this used paying, contracted customers (there is no way for you to know this, unless you work for one 😉); however... I should have some ways of finding this out...
* is this used by very large third party vendors

zsd4yr commented on src/System.Windows.Forms/src/System/Windows/Forms/ColumnClickEvent.cs:
```diff
@@ -1,38 +0,0 @@
-    ///
-    ///    </para>
-    /// </devdoc>
-    public class ColumnClickEventArgs : EventArgs {
```
Same concerns as with CacheVirtualItemsEventArgs. This may be the case with any such public class deletions. I'll try to get a definitive answer here.

zsd4yr commented on src/System.Windows.Forms/src/System/Windows/Forms/ColumnClickEventArgs.cs:
```diff
@@ -0,0 +1,20 @@
+    /// Provides data for the <see cref='System.Windows.Forms.ListView.OnColumnClick'/>
+    /// event.
+    /// </devdoc>
+    public class ColumnClickEventArgs : EventArgs
```
Was this class present in another file and you've put it in its own? Which file?

If not, why would you add it?

zsd4yr commented on src/System.Windows.Forms/src/System/Windows/Forms/ColumnWidthChangedEvent.cs:
```diff
@@ -1,34 +0,0 @@
-    /// <include file='doc\ColumnWidthChangedEvent.uex' path='docs/doc[@for="ColumnWidthChangedEventArgs"]/*' />
-    /// <devdoc>
-    /// </devdoc>
-    public class ColumnWidthChangedEventArgs : EventArgs {
```
Same as CacheVirtualItemsEventArgs concern; marking for my own purposes

zsd4yr commented on src/System.Windows.Forms/src/System/Windows/Forms/ColumnWidthChangedEventArgs.cs:
```diff
@@ -0,0 +1,16 @@
+
+namespace System.Windows.Forms
+{
+    public class ColumnWidthChangedEventArgs : EventArgs
```
Yeah did you add this one too? Sorry --there's a lot to look through here

zsd4yr commented on src/System.Windows.Forms/src/System/Windows/Forms/ColumnWidthChangingEvent.cs:
```diff
@@ -1,55 +0,0 @@
-using System.ComponentModel;
-
-    /// <include file='doc\ColumnWidthChangingEvent.uex' path='docs/doc[@for="ColumnWidthChangingEventArgs"]/*' />
-    public class ColumnWidthChangingEventArgs : CancelEventArgs {
```
Same as CacheVirtualItemsEventArgs concern; marking for my own purposes

zsd4yr commented:
Hey @hughbe, after looking through your changes here, I'm finding some equivalence classes of changes I'd like to break out and discuss separately:
1. code cleanup --removing using statements, tabs, spaces, etc... these should be fine assuming everything builds etc...
1. changing getters / setters --again, this should be fine so long as code paths persist
1. changing variable names (capitalization) --this may be a problem if the classes are serialized or if these are public fields (see below point about new APIs) 
1. removing "dead" public APIs--this will likely require more overhead, as I noted above
1. adding new APIs/ events -- I first want to clarify with you if you have done this, or if any new classes are just the result of you copy / pasting those out of other code files. If you are going to add new public APIs, even for events, this is going to require API review.

zsd4yr commented on src/System.Windows.Forms/src/System/Windows/Forms/DrawListViewColumnHeaderEventArgs.cs:
```diff
@@ -2,193 +2,125 @@
         /// </devdoc>
-        public void DrawBackground() {
-            if (Application.RenderWithVisualStyles) {
-                VisualStyleRenderer vsr = new VisualStyleRenderer(VisualStyleElement.Header.Item.Normal);
```
we may have to discuss this change to var --unsure what the best strategy is here, and the team has yet to conclude a consistent semantic

hughbe commented:
By the way I have not deleted or added any public api - I think you’re reviewing all the changes rather than commit by commit. If you look at commit 1, then I’ve renamed all the files to match their actual name. Then the cleanup in commit 2 just demonstrates the cleanup. I’m guessing a limitation of github’s interface suggests that I’ve deleted and created new files

Will look into extracting changes

sharwell commented:
> Updated the PR only to rename the files. 

👍

sharwell commented:
📝 Once the entire solution conforms to this pattern, we can enabled the StyleCop Analyzers rule that checks for consistency during builds.

zsd4yr commented:
hey @hughbe unfortunately, even though we don't use this API, our API review board has determine that it is used by a sufficiently high percentage of user applications, which means we cannot in good faith remove it. Sorry you went to all the trouble. 

This kind of thing may also occur elsewhere.

hughbe commented:
I’m very confused - this PR doesn’t change anything in the binary? This just literally changed the name of the source code file...

zsd4yr commented:
Ah wait I see, that's right we moved that stuff didn't we... damn 1 sec...

zsd4yr commented:
good catch, sorry
